src/plugins/vis_default_editor/public/components/agg_params_helper.ts

```diff
@@ -115,6 +115,7 @@ function getAggParamsToRender({
     }
 
     if (!paramEditor) {
+      if (agg.type.name === 'count' && param.type === 'json') return;
```

Comment:
Your solutions was tested locally and it works fine. But could you please use `METRIC_TYPES.COUNT` instead of hardcoded value. 

Also from the design perspective I don't like idea to add some `if`'s into high level component. From my point of view all of that edge cases should be handled in `src/plugins/vis_default_editor/public/components/agg_params_map.ts`

But looks like `agg_params_map` also requires some refactoring to exclude some fields from  that map =( but I prefer to go this way

@sulemanof  what do you think?

Reply:
In its simplest form, it looks like

agg_params_map.ts: 

![image](https://user-images.githubusercontent.com/20072247/90626276-6685d100-e223-11ea-9f08-ea1b3a011ad5.png)

agg_params_helper.ts:
![image](https://user-images.githubusercontent.com/20072247/90626443-a5b42200-e223-11ea-8251-1df5584e2770.png)

Reply:
IMHO,
the restriction of an any param should come from an agg type.
The source of building any aggregation on UI is the `agg.type.params`, which is based on agg config.
In that case, we should restrict the `count` metric have the `json` param in its config.
I could suggest smth like that:

`src/plugins/data/common/search/aggs/metrics/count.ts`

![image](https://user-images.githubusercontent.com/31325372/90634157-ef0a6e80-e22f-11ea-821d-b5d33e2e8f23.png)

and than in `src/plugins/data/common/search/aggs/agg_type.ts`

![image](https://user-images.githubusercontent.com/31325372/90634295-2416c100-e230-11ea-86c1-915abd1d56d9.png)

The same is happening now to `customLabel` param:

![image](https://user-images.githubusercontent.com/31325372/90634403-4c062480-e230-11ea-8b16-859ab761d1d7.png)

Reply:
Probably yes, `/plugins/data/common/search/aggs/metrics/count_fn.ts` should be also updated

Reply:
Ok @alexwizp, let's sync with @sulemanof and I will proceed with the solution after. Maybe a minor refactor is needed here.

Reply:
Than you both ❤️ Will look into this